server/lib/pbench/process_tb.py

```diff
@@ -0,0 +1,108 @@
+import os
+import glob
+import subprocess
+
+from collections import namedtuple
+from logging import Logger
+from pathlib import Path
+
+from pbench import PbenchConfig
+
+Results = namedtuple("Results", ["nstatus", "ntotal", "ntbs"])
+
+
+class ProcessTb:
+    def __init__(self, config: PbenchConfig, logger: Logger):
+        """ Processes Tar ball, gets the tar ball from receiving directory
+        and copies it to the remote server using HTTP PUT method.
+
+            Args -
+                config -- PbenchServer config object
+                logger -- logger object to use when emitting log entries during
+                            operation
+
+        """
+        self.config = config
+        self.logger = logger
+
+    def get_receive_dir(self) -> Path:
+        receive_dir_prefix = self.config.get(
+            "pbench-server", "pbench-receive-dir-prefix"
+        )
+        if not receive_dir_prefix:
+            self.logger.error(
+                "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
+            )
+            return None
+
+        receive_dir = Path(f"{receive_dir_prefix}-002").resolve()
+        if not receive_dir.is_dir():
+            self.logger.error(
+                "Failed: {} does not exist, or is not a directory", receive_dir
+            )
+            return None
+
+        return receive_dir
+
+    def run_subprocess(self, controller: str, tb: str, token: str):
+        """Runs `pbench-results-push` command with controller, tb
+        and token as options"""
+
+        subprocess.run(["pbench-results-push", controller, tb, token])
+
+    def process_tb(self, receive_dir: Path) -> Results:
+        """Receives a tar ball and copies it to the remote server
+            using the server's HTTP PUT method
+
+            Args -
+                receive_dir -- directory where tarballs are received from agent
+
+        """
+
+        # Check for results that are ready for processing: version 002 agents
+        # upload the MD5 file as xxx.md5.check and they rename it to xxx.md5
+        # after they are done with MD5 checking so that's what we look for.
+        list_check = glob.glob(
+            os.path.join(receive_dir, "**", "*.tar.xz.md5"), recursive=True
+        )
+
+        self.logger.info("{}", self.config.TS)
+        list_check.sort()
+        nstatus = ""
+
+        ntotal = ntbs = 0
+
+        token = self.config.get("pbench-server", "put-token")
+
+        for tbmd5 in list_check:
+            ntotal += 1
+
+            # Extracts full pathname of tarball from the name of hashfile
+            # by trimming .md5 suffix
+            tb = Path(tbmd5[0 : -len(".md5")])
```

Comment:
I will have to look into this as it doesn't seem to work with string as well as a Path variable. Maybe I am missing something.

Reply:
I'm sorry -- I've possibly made several blunders here.

First, I suggested elsewhere a change which means that `tbmd5` is now a `Path` instead of a `str`, and my suggestion here won't work if `tbmd5` is a `Path` -- if that's the case, then you need to stringize it:
```
            tb = Path(str(tbmd5).removesuffix(".md5"))
```

Lastly, I believe that `removesuffix()`, as sexy as it is, is not available for our use on the `b0.69` branch, because it requires Python 3.9, and V0.69 only requires Python 3.8.  😿 

So, nevermind.

Reply:
yeah, that must have been the issue, because even after changing the path into the string it didn't work.